### PR TITLE
Docsum

### PR DESCRIPTION
**Pull request by MSCetin37**
## Description

Extend the current Document Summarization Application by incorporating video and audio summary features. This enhancement will enable the application to summarize video and audio content in addition to text documents, thereby broadening its utility and applicability.

## Issues
https://github.com/opea-project/docs/blob/main/community/rfcs/24-06-21-OPEA-001-DocSum_Video_Audio.md

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [x] Breaking change (fix or feature that would break existing design and interface)
- [ ] Others (enhancement, documentation, validation, etc.)

## Dependencies
N/A

## Tests
- GenAIExamples/DocSum/tests/test_compose_on_gaudi.sh  
- GenAIExamples/DocSum/tests/test_compose_on_xeon.sh  
- GenAIExamples/DocSum/tests/test_gmc_on_gaudi.sh  
- GenAIExamples/DocSum/tests/test_gmc_on_xeon.sh  
- GenAIExamples/DocSum/tests/test_manifest_on_gaudi.sh  
- GenAIExamples/DocSum/tests/test_manifest_on_xeon.sh

### CODE REVIEW

**Review by ashahba — CHANGES_REQUESTED** [review]
Hi @MSCetin37 
Thanks for this PR!
There are some conflicts that need to be resolved first and prior to review.
The PR shows nearly `100` files have changed, which is very unlikely and due to unsuccessful  sync with main branch.

**Comment by ashahba** [comment]
This PR is dependant on: https://github.com/opea-project/GenAIComps/pull/865 and should wait until that PR is merged.

File: DocSum/README.md
Comment (eero-t) [inline DocSum/README.md:156]: Shouldn't the other parameters be also given, at least language?

And because this is "document summary" application, I would really like example of an actual document (file) being summarized, instead of some ridiculously short text.

(This comment applies to all docsum usage examples in this PR, not just the one in this README.)

File: DocSum/docker_compose/intel/cpu/xeon/README.md
Comment (eero-t) [inline DocSum/docker_compose/intel/cpu/xeon/README.md:93]:
```suggestion
   export HUGGINGFACEHUB_API_TOKEN="Your_HuggingFace_API_Token"
```

(This comment applies to all the READMEs in this PR.)

File: DocSum/docker_compose/intel/hpu/gaudi/README.md
Comment (eero-t) [inline DocSum/docker_compose/intel/hpu/gaudi/README.md:64]:
```suggestion
Default model is "Intel/neural-chat-7b-v3-3". Change "LLM_MODEL_ID" environment variable in commands below if you want to use another model.
When using gated models, you also need to provide [HuggingFace token](https://huggingface.co/docs/hub/security-tokens) to "HUGGINGFACEHUB_API_TOKEN" environment variable.
```

(This comment applies also to all other READMEs in this PR.)

File: DocSum/README.md
Comment (MSCetin37) [reply DocSum/README.md:156]: Thanks for your feedback. 
You can upload pdf, doc ,docx, audio and video documents. 

image

Parameter (language, llm_model, whisper model etc.) setting will be included in v.1.2

Comment (eero-t) [reply DocSum/README.md:156]: That's the UI, I'm not really interested about that, but the megaservice API for uploading such files...

File: DocSum/docker_compose/intel/cpu/xeon/README.md
Comment (MSCetin37) [reply DocSum/docker_compose/intel/cpu/xeon/README.md:93]: Your suggestion has a valid point, but this is how it is used in OPEA (across 21 different README files). To maintain consistency, it should be changed throughout the entire OPEA, which is not within the scope of this PR

File: DocSum/docker_compose/intel/hpu/gaudi/README.md
Comment (MSCetin37) [reply DocSum/docker_compose/intel/hpu/gaudi/README.md]: Thank you for your suggestion. Your comment also applies to all other README files in this PR

**Comment by chensuyue** [comment]
If this is target v1.1, please make sure it can be merged before 2024/11/15.

File: DocSum/ui/gradio/Dockerfile
Comment (chensuyue) [inline DocSum/ui/gradio/Dockerfile:1]: Please align the code structure.

File: DocSum/docker_image_build/build.yaml
Comment (chensuyue) [inline DocSum/docker_image_build/build.yaml:11]: In the docker build yaml, keep the service name the same as docker image name.   
Please also add the new ui docker image build here.

File: DocSum/docker_compose/intel/cpu/xeon/README.md
Comment (XinyaoWa) [inline DocSum/docker_compose/intel/cpu/xeon/README.md:199]: Please add the guideline for how to query with video/image/etc. with megaservice

File: DocSum/tests/test_compose_on_gaudi.sh
Comment (XinyaoWa) [inline DocSum/tests/test_compose_on_gaudi.sh:212]: Please add the validation for megaservice and frontend for all types(image/video/etc...)

**Comment by XinyaoWa** [comment]
@WenjiaoYue  Please check the UI deletion

**Comment by XinyaoWa** [comment]
> @WenjiaoYue Please check the UI deletion

Please keep the original UI codes and make them runnable.

**Comment by chensuyue** [comment]
Does this PR fix this issue? https://github.com/opea-project/GenAIExamples/issues/1129

File: DocSum/tests/test_manifest_on_xeon.sh
Comment (chensuyue) [inline DocSum/tests/test_manifest_on_xeon.sh:8]: why edit this part?

File: DocSum/tests/test_manifest_on_xeon.sh
Comment (chensuyue) [inline DocSum/tests/test_manifest_on_xeon.sh:16]: This is not needed anymore.

File: DocSum/tests/test_manifest_on_xeon.sh
Comment (chensuyue) [inline DocSum/tests/test_manifest_on_xeon.sh:24]: we should not edit this, this will cause CI issue.

File: DocSum/tests/test_compose_on_gaudi.sh
Comment (chensuyue) [inline DocSum/tests/test_compose_on_gaudi.sh:61]: Remove this line please!! This will clean up all the images in CI machine, including the image registry. The CI machine is shared by a list of projects and tests, so this will definitely impact other test!

File: DocSum/tests/test_compose_on_gaudi.sh
Comment (chensuyue) [inline DocSum/tests/test_compose_on_gaudi.sh:229]: There is already docker stop before the test.

Comment (MSCetin37) [reply DocSum/tests/test_compose_on_gaudi.sh:240]: it is removed

File: DocSum/tests/test_compose_on_gaudi.sh
Comment (MSCetin37) [reply DocSum/tests/test_compose_on_gaudi.sh]: it is removed

File: DocSum/tests/test_manifest_on_xeon.sh
Comment (MSCetin37) [reply DocSum/tests/test_manifest_on_xeon.sh]: fixed

File: DocSum/tests/test_manifest_on_xeon.sh
Comment (MSCetin37) [reply DocSum/tests/test_manifest_on_xeon.sh]: fixed

Comment (MSCetin37) [reply DocSum/tests/test_manifest_on_xeon.sh]: fixed

File: DocSum/tests/test_manifest_on_xeon.sh
Comment (MSCetin37) [reply DocSum/tests/test_manifest_on_xeon.sh]: fixed

File: DocSum/tests/test_compose_on_gaudi.sh
Comment (chensuyue) [inline DocSum/tests/test_compose_on_gaudi.sh:5]: why comments this?

File: DocSum/tests/test_compose_on_xeon.sh
Comment (chensuyue) [inline DocSum/tests/test_compose_on_xeon.sh:62]: why use 20 instead of 100, it usually take several minutes to start the service.

File: DocSum/tests/test_compose_on_gaudi.sh
Comment (MSCetin37) [reply DocSum/tests/test_compose_on_gaudi.sh:5]: uncommented

File: DocSum/tests/test_compose_on_xeon.sh
Comment (MSCetin37) [reply DocSum/tests/test_compose_on_xeon.sh]: set to 100

**Comment by MSCetin37** [comment]
Depends on this repo: https://github.com/opea-project/GenAIComps/pull/902

File: DocSum/docker_compose/intel/hpu/gaudi/compose.yaml
Comment (ashahba) [inline DocSum/docker_compose/intel/hpu/gaudi/compose.yaml:65]: Is this a typo? or we want to use different container and local ports?

File: DocSum/tests/test_compose_on_xeon.sh
Comment (ashahba) [inline DocSum/tests/test_compose_on_xeon.sh:104]: This line is not needed

File: DocSum/ui/docker/Dockerfile.gradio
Comment (ashahba) [inline DocSum/ui/docker/Dockerfile.gradio:20]: Sort this list and remove `vim`

File: DocSum/README.md
Comment (ashahba) [inline DocSum/README.md:160]: Leave one space after `#`

File: DocSum/README.md
Comment (ashahba) [inline DocSum/README.md:185]: Remove this line

**Review by ashahba — CHANGES_REQUESTED** [review]

**Review by ashahba — APPROVED** [review]
LGTM!

File: DocSum/docker_compose/intel/cpu/xeon/README.md
Comment (XinyaoWa) [inline DocSum/docker_compose/intel/cpu/xeon/README.md:213]: Could you please add the complete usage for megaservice here, as the user will only look into this readme, while UT is designed just for developer, not for users.

File: DocSum/tests/test_compose_on_gaudi.sh
Comment (XinyaoWa) [inline DocSum/tests/test_compose_on_gaudi.sh:139]: Could you please revert the frontend UT?

File: DocSum/docker_image_build/build.yaml
Comment (lianhao) [inline DocSum/docker_image_build/build.yaml:21]: Here we change the content of a published docker image to something complete different, svelte UI -> gradio UI, without using a new image name. I'm wondering this is the right way to do.

File: DocSum/docsum.py
Comment (lianhao) [inline DocSum/docsum.py:27]: Can we make all multimedia related things optional?  This kind of last minute changes which introduces new required components will break the existing K8S(and helm chart) deployment, and caught us in surprise. We need time to add new components support in k8s(i.e. a2t, v2a, multimedia2text).

**Comment by lianhao** [comment]
My previous comments somehow got hided, so let me put here:

- https://github.com/opea-project/GenAIExamples/pull/1095#discussion_r1843226642 
- https://github.com/opea-project/GenAIExamples/pull/1095#discussion_r1843232401

File: DocSum/docker_image_build/build.yaml
Comment (lianhao) [inline DocSum/docker_image_build/build.yaml:53]: In GenAIComps [related PR](https://github.com/opea-project/GenAIComps/pull/865/files#diff-022171efe111a160f35a88b9872634570129a881f83bdf3780784499884dcf2d), the image is named `dataprep-audio2text`, we should have both the image names synced.

File: DocSum/docker_image_build/build.yaml
Comment (lianhao) [inline DocSum/docker_image_build/build.yaml:43]: In GenAIComps [related PR](https://github.com/opea-project/GenAIComps/pull/865/files#diff-022171efe111a160f35a88b9872634570129a881f83bdf3780784499884dcf2d), the image is named `dataprep-multimedia2text`, we should have both the image names synced.

File: DocSum/docker_image_build/build.yaml
Comment (lianhao) [inline DocSum/docker_image_build/build.yaml:63]: In GenAIComps [related PR](https://github.com/opea-project/GenAIComps/pull/865/files#diff-022171efe111a160f35a88b9872634570129a881f83bdf3780784499884dcf2d), the image is named `dataprep-video2audio`, we should have both the image names synced.

**Review by lianhao — CHANGES_REQUESTED** [review]

File: DocSum/docker_image_build/build.yaml
Comment (chensuyue) [reply DocSum/docker_image_build/build.yaml:53]: Let's use the same image name, or there will be duplicate images released.

File: DocSum/docker_image_build/build.yaml
Comment (chensuyue) [reply DocSum/docker_image_build/build.yaml:43]: Thanks @lianhao to point this, let's unify the image name.

File: DocSum/docker_image_build/build.yaml
Comment (chensuyue) [reply DocSum/docker_image_build/build.yaml:63]: Same as before.

File: DocSum/docker_image_build/build.yaml
Comment (chensuyue) [reply DocSum/docker_image_build/build.yaml:21]: Yes, this is not allowed. Do not delete other images. Just create a new one.

File: DocSum/docker_compose/intel/cpu/xeon/README.md
Comment (dmsuehir) [inline DocSum/docker_compose/intel/cpu/xeon/README.md:71]: The way it's written I am getting a build error. I think this needs to be changed back to navigating to the `ui` folder and then using `-f docker/Dockerfile.gradio`
```suggestion
cd GenAIExamples/DocSum/ui
```

My docker build error is:
```
 => ERROR [4/7] COPY ./gradio/docsum_ui_gradio.py /home/user/docsum_ui_gradio.py                                                                                                                                                                                                             0.0s
 => ERROR [5/7] COPY ./gradio/requirements.txt /home/user/requirements.txt                                                                                                                                                                                                                   0.0s
------
 > [4/7] COPY ./gradio/docsum_ui_gradio.py /home/user/docsum_ui_gradio.py:
------
------
 > [5/7] COPY ./gradio/requirements.txt /home/user/requirements.txt:
------
Dockerfile.gradio:21
--------------------
  19 |     # Copy the application code and requirements file to the container
  20 |     COPY ./gradio/docsum_ui_gradio.py /home/user/docsum_ui_gradio.py
  21 | >>> COPY ./gradio/requirements.txt /home/user/requirements.txt 
  22 |     
  23 |     # Install Python dependencies
--------------------
ERROR: failed to solve: failed to compute cache key: failed to calculate checksum of ref e4041a4d-34f1-4db4-a5ee-a6cb16701b70::z6z1y6myra817r1nrl9p15cvs: "/gradio/requirements.txt": not found
```

File: DocSum/docker_compose/intel/cpu/xeon/README.md
Comment (dmsuehir) [inline DocSum/docker_compose/intel/cpu/xeon/README.md:72]:
```suggestion
docker build -t opea/docsum-ui:latest --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy -f docker/Dockerfile.gradio .
```

File: DocSum/docker_image_build/build.yaml
Comment (chensuyue) [inline DocSum/docker_image_build/build.yaml:20]: Why did you remove this image?  
And I didn't see the new image `docsum-gradio-ui`, if the image not in this file, it won't be released on docker hub.

File: DocSum/docker_image_build/build.yaml
Comment (chensuyue) [inline DocSum/docker_image_build/build.yaml:22]: Use the correct dockerfile name, 
image

**Review by lvliang-intel — APPROVED** [review]

**Pull request merged**
